src/Resources/skeleton/forgottenPassword/ForgottenPasswordController.tpl.php

```diff
@@ -0,0 +1,118 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace; ?>;
+
+use <?= $user_full_class_name ?>;
+use <?= $username_form_full_class_name ?>;
+use <?= $resetting_form_full_class_name ?>;
+use Symfony\Bundle\FrameworkBundle\Controller\<?= $parent_class_name; ?>;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+class <?= $class_name; ?> extends <?= $parent_class_name; ?><?= "\n" ?>
+{
+    // token sent by e-mail is valid 24 hours
+    const TOKEN_TTL_HOURS = 24;
+
+    /**
+     * @Route("/forgotten-password", name="app_forgotten_password")
+     */
+    public function request(Request $request, \Swift_Mailer $mailer): Response
+    {
+        $form = $this->createForm(<?= $username_form_class_name ?>::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
+                '<?= $username_field ?>' => $form->get('<?= $username_field ?>')->getData(),
+            ]);
+
+            // do not reveal whether an user account was found or not
+            if (!$user) {
+                return $this->redirectToRoute('app_check_email');
+            }
+
+            // generate a reset password token, the user could use to change his password
+            $resetPasswordToken =$this->generateConfirmationToken();
+
+            $user->setResetPasswordToken($resetPasswordToken);
+            $user->setPasswordRequestedAt(new \DateTime());
```

Comment:
This is probably a spot where I'd prefer to not generate setter methods for the 2 new columns, and instead have something like: `$user->generatePasswordResetToken()` (which we would of course generate for them) that takes care of setting both of these fields.